raster-foundry/raster-foundry
# Add access logging to backsplash

## Overview

This PR adds access logging to backsplash, if you turn it on with the `BACKSPLASH_SERVER_ACCESS_LOGGING` environment variable.

### Checklist


### Notes

This grabs everything mentioned in the issue and a little more, namely the Amazon trace ID, anything present from an ad hoc headers whitelist, and the status code. I threw everything into a flat json because I have a vague recollection that Athena doesn't love nested json for queries, so headers and non-headers are present at the same level of the json. There are also other things that are present in the requests, but I can't see what they are because `Vault`s (the type backing the [`attributes`](https://github.com/http4s/http4s/blob/main/core/src/main/scala/org/http4s/Message.scala#L46) of `Request`s) [don't expose a way to see what's inside them](https://github.com/ChristopherDavenport/vault/blob/master/core/src/main/scala/io/chrisdavenport/vault/Vault.scala).

Also it removes some stuff that wasn't needed anymore, like some config keys and the metric middleware.

## Testing Instructions

- assemble backsplash server
- bring up your server
- make sure the tile visibility for project `7e584c31-f5d1-4a02-9428-e83006642375` is `PUBLIC`
- send a request from the command line: `http http://localhost:8081/7e584c31-f5d1-4a02-9428-e83006642375/17/33613/52009?foo=bar`
- look at the logs that you get -- they should make sense
- fire up Groundwork and look at some more tiles and look at the logs -- they should have more info than the cli request, since browsers attach stuff
- check the request headers from teh browser and see if there's anything else you wish was included

Closes #5528
